### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by MariaShapovalova09**

### CODE REVIEW

File: ShapovalovaM_transformers.ipynb [hunk 1]
```diff
@@ -0,0 +1,1998 @@
+{
```
Comment (dpaleyev) [inline ShapovalovaM_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.            self.cls_token = nn.Parameter(torch.randn(1, 1, embed_dim))
</pre><p>Тут можно так делать, но не обязательно. Вообще при малом количестве данных обучаемый эмбединг для cls токены может привести к переобучению. Но в больших моделях обучают, как ты написала</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MariaShapovalova09/ML_summer/pull/1/discussion/#comment-2188135606'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (dpaleyev) [inline ShapovalovaM_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #34.            self.dropout = nn.Dropout(dropout)
</pre><p>Не просили делать дропаут здесь. Тоже может ухудшить обучение. В целом не стоит добавлять свои идеи в готовые и работающие архитектуры :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MariaShapovalova09/ML_summer/pull/1/discussion/#comment-2188135607'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (dpaleyev) [inline ShapovalovaM_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #48.            K_proj = torch.einsum('pd,bhdk-&gt;bhpk', self.e_proj, k)
</pre><p>жесть, можно было просто сделать self.e_proj @ k</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MariaShapovalova09/ML_summer/pull/1/discussion/#comment-2188135608'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='81'/>

Comment (dpaleyev) [inline ShapovalovaM_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #64.    class LinformerBlock(EncoderBlock):
</pre><p>Непонятно а зачем наследовать было если ты заново написала forward блок (и это копипаст). Ты могла отнаследоваться заменить блок <strong>init</strong> а форвард оставить тем же если не меняла нейминги </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MariaShapovalova09/ML_summer/pull/1/discussion/#comment-2188135609'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='81'/>
